test/integration/noderesourcetopology_test.go

```diff
@@ -622,6 +628,319 @@ func TestTopologyMatchPlugin(t *testing.T) {
+			pods: []*v1.Pod{
+				util.WithLimits(
+					util.WithLimits(
+						st.MakePod().Namespace(ns).Name(testPodName).SchedulerName(leastNUMAScheduler),
+						map[string]string{cpu: "2", memory: "4Gi"}, false),
+					map[string]string{cpu: "2", memory: "6Gi"}, false).Obj(),
+			},
+
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				MakeNRT().Name("fake-node-1").Policy(topologyv1alpha1.BestEffortContainerLevel).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						}).Obj(),
+				MakeNRT().Name("fake-node-2").Policy(topologyv1alpha1.BestEffortContainerLevel).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "3Gi"),
+						}).Obj(),
+			},
+			expectedNodes: []string{"fake-node-1", "fake-node-2"},
+		},
+		{
+			name: "Scheduling Guaranteed pod with init container with LeastNUMANodes strategy scheduler and container scope",
+			pods: []*v1.Pod{
+				util.WithLimits(
+					util.WithLimits(
+						st.MakePod().Namespace(ns).Name(testPodName).SchedulerName(leastNUMAScheduler),
+						map[string]string{cpu: "2", memory: "4Gi"}, false),
+					map[string]string{cpu: "2", memory: "6Gi"}, true).Obj(),
+			},
+
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				MakeNRT().Name("fake-node-1").Policy(topologyv1alpha1.BestEffortContainerLevel).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						}).Obj(),
+				MakeNRT().Name("fake-node-2").Policy(topologyv1alpha1.BestEffortContainerLevel).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "3Gi"),
+						}).Obj(),
+			},
+			expectedNodes: []string{"fake-node-1", "fake-node-2"},
```

Comment:
The slice is sorted from the less desirable (leftmost) to most desirable (rightmost) right?

Reply:
What do you mean by desirable? Both nodes will have the same score(requires same amount of NUMA nodes to satisfy resource requests).

Reply:
There are couple of those:
* `Scheduling Guaranteed pod with LeastNUMANodes strategy scheduler, different allocation pod scope`
* `Scheduling Guaranteed multi container pod with LeastNUMANodes strategy scheduler and pod scope`
* `Scheduling Guaranteed pod with multiple containers with LeastNUMANodes strategy scheduler, different resource allocation, pod scope`
* `Scheduling Guaranteed pod with multiple containers with LeastNUMANodes strategy scheduler, different resource allocation, container scope`

Only one test is checking the case where a pod can't fit on a node:
* `Scheduling Guaranteed multiple containers pod with LeastNUMANodes strategy scheduler, can't fit one node, pod scope`

Reply:
Thanks! what I would like to have covered is a case on which the score is actually different from 2+ compute nodes without interference from NodeResourceTopoologyMatch `filter` . Unless I'm mistaken, 
in `Scheduling Guaranteed pod with LeastNUMANodes strategy scheduler, different allocation pod scope`
the pod here can't fit on either NUMA zones on fake-node-1 (requests too many CPUs), so the filter is ruling it out, so the score would only score fake-node-2
same for `Scheduling Guaranteed multi container pod with LeastNUMANodes strategy scheduler and pod scope` and few others.
So existing integration tests looks good, I would just like to see 1+ tests on which the score is the deciding factor without being constrained by `filter` into a obvious decision.

I reckon that creating such condition can be nontrivial, exactly because we should NOT disable `filter` if possible

Reply:
I think `filter` plugins only works for `SingleNUMANode` policies and is noop for other policies, so in this example it's won't interfere.

Reply:
Yeah, after passing `-v=5` to tests we can see:
```sh
=== RUN   TestTopologyMatchPlugin/Scheduling_Guaranteed_pod_with_multiple_containers_with_LeastNUMANodes_strategy_scheduler,_different_resource_allocation,_pod_scope
    noderesourcetopology_test.go:1173: Start-topology-match-test Scheduling Guaranteed pod with multiple containers with LeastNUMANodes strategy scheduler, different resource allocation, pod scope
    noderesourcetopology_test.go:1183: Creating Pod "topology-aware-scheduler-pod"
I0124 13:17:34.911563 2433617 eventhandlers.go:116] "Add event for unscheduled pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:34.911619 2433617 scheduling_queue.go:957] "About to try and schedule pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:34.911636 2433617 schedule_one.go:85] "Attempting to schedule pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:34.911749 2433617 passthrough.go:38] "Lister for nodeResTopoPlugin" lister=&{indexer:0xc000536108}
I0124 13:17:34.911771 2433617 filter.go:209] "Found NodeResourceTopology" nodeTopology="fake-node-1"
I0124 13:17:34.911779 2433617 passthrough.go:38] "Lister for nodeResTopoPlugin" lister=&{indexer:0xc000536108}
I0124 13:17:34.911806 2433617 filter.go:217] "Policy handler not found" policy="BestEffortPodLevel"
I0124 13:17:34.911812 2433617 filter.go:209] "Found NodeResourceTopology" nodeTopology="fake-node-2"
I0124 13:17:34.911827 2433617 filter.go:217] "Policy handler not found" policy="BestEffortPodLevel"
I0124 13:17:34.911918 2433617 passthrough.go:38] "Lister for nodeResTopoPlugin" lister=&{indexer:0xc000536108}
I0124 13:17:34.911929 2433617 passthrough.go:38] "Lister for nodeResTopoPlugin" lister=&{indexer:0xc000536108}
I0124 13:17:34.912130 2433617 default_binder.go:52] "Attempting to bind pod to node" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod" node="fake-node-2"
I0124 13:17:34.915339 2433617 cache.go:402] "Finished binding for pod, can be expired" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:34.915400 2433617 schedule_one.go:266] "Successfully bound pod to node" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod" node="fake-node-2" evaluatedNodes=2 feasibleNodes=2
I0124 13:17:34.915834 2433617 eventhandlers.go:159] "Delete event for unscheduled pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:34.915847 2433617 eventhandlers.go:184] "Add event for scheduled pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
    noderesourcetopology_test.go:1200: Pod topology-aware-scheduler-pod scheduled
    noderesourcetopology_test.go:1207: Pod "topology-aware-scheduler-pod" is on a nodes as expected.
    noderesourcetopology_test.go:1234: Case Scheduling Guaranteed pod with multiple containers with LeastNUMANodes strategy scheduler, different resource allocation, pod scope finished
I0124 13:17:35.923940 2433617 eventhandlers.go:204] "Update event for scheduled pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:35.925712 2433617 eventhandlers.go:229] "Delete event for scheduled pod" pod="integration-test-8149a00c-fb72-4a0d-b849-53d6277354b3/topology-aware-scheduler-pod"
I0124 13:17:35.938264 2433617 nrtutils.go:136] init scheduler success
```
and line `I0124 13:17:34.911827 2433617 filter.go:217] "Policy handler not found" policy="BestEffortPodLevel"` means the `filter` plugin is not being triggered for this pod

Reply:
ok, I was at fault: I was misremembering what `expectedNodes` means in this context, so the above tests are indeed good and cover the cases I was hinting. It seems we're all good.